sdk/metric/aggregator/test/test.go

```diff
@@ -75,66 +80,64 @@ func (m *metricBatcher) Export(context.Context, export.MetricRecord, export.Metr
 }
 
 func RunProfiles(t *testing.T, f func(*testing.T, Profile)) {
-	for _, profile := range profiles {
+	for _, profile := range newProfiles() {
 		t.Run(profile.NumberKind.String(), func(t *testing.T) {
 			f(t, profile)
 		})
 	}
 }
 
-type Numbers []core.Number
+type Numbers struct {
+	kind    core.NumberKind
+	numbers []core.Number
+}
+
+func NewNumbers(kind core.NumberKind) Numbers {
+	return Numbers{
+		kind: kind,
+	}
+}
+
+func (n *Numbers) Append(v core.Number) {
+	n.numbers = append(n.numbers, v)
+}
 
 func (n *Numbers) Sort() {
 	sort.Sort(n)
 }
 
 func (n *Numbers) Less(i, j int) bool {
-	return (*n)[i] < (*n)[j]
+	return n.numbers[i].CompareNumber(n.kind, n.numbers[j]) < 0
 }
 
 func (n *Numbers) Len() int {
-	return len(*n)
+	return len(n.numbers)
 }
 
 func (n *Numbers) Swap(i, j int) {
-	(*n)[i], (*n)[j] = (*n)[j], (*n)[i]
+	n.numbers[i], n.numbers[j] = n.numbers[j], n.numbers[i]
 }
 
-func (n *Numbers) Sum(kind core.NumberKind) core.Number {
+func (n *Numbers) Sum() core.Number {
 	var sum core.Number
-	for _, num := range *n {
-		sum.AddNumber(kind, num)
+	for _, num := range n.numbers {
+		sum.AddNumber(n.kind, num)
 	}
 	return sum
 }
 
 func (n *Numbers) Count() int64 {
-	return int64(len(*n))
+	return int64(len(n.numbers))
 }
 
-func (n *Numbers) Median(kind core.NumberKind) core.Number {
-	if !sort.IsSorted(n) {
-		panic("Sort these numbers before calling Median")
-	}
-
-	l := len(*n)
-	if l%2 == 1 {
-		return (*n)[l/2]
-	}
-
-	lower := (*n)[l/2-1]
-	upper := (*n)[l/2]
+func (n *Numbers) Min() core.Number {
+	return n.numbers[0]
+}
 
-	sum := lower
-	sum.AddNumber(kind, upper)
+func (n *Numbers) Max() core.Number {
+	return n.numbers[len(n.numbers)-1]
+}
 
-	switch kind {
-	case core.Uint64NumberKind:
-		return core.NewUint64Number(sum.AsUint64() / 2)
-	case core.Int64NumberKind:
-		return core.NewInt64Number(sum.AsInt64() / 2)
-	case core.Float64NumberKind:
-		return core.NewFloat64Number(sum.AsFloat64() / 2)
-	}
-	panic("unknown number kind")
+func (n *Numbers) Median() core.Number {
+	return n.numbers[len(n.numbers)/2]
```

Comment:
why is media formula changed?

Reply:
I'm using Median() here to mean Quantile(0.5), and there are too many definitions for Quantile to list. (I found http://mathworld.wolfram.com/Quantile.html informative)

After some research in the metrics space, I concluded that the most frequent definition for a quantile returns the smallest value from the original data set which is >= the given quantile. This means we always return a value in the set, not an interpolated value.

Interpolation has risks -- it makes sense when you have a data model, but we don't in this case. If we know the data is uniform or gaussian, then interpolation could make sense.

Reply:
I added  "// Median() is an alias for Quantile(0.5)." and
```
	// Note that len(n.numbers) is 1 greater than the max element
	// index, so dividing by two rounds up.  This gives the
	// intended definition for Quantile() in tests, which is to
	// return the smallest element that is at or above the
	// specified quantile.
```